src/main/scala/mesosphere/marathon/core/instance/update/InstanceUpdater.scala

```diff
@@ -27,14 +27,9 @@ object InstanceUpdater extends StrictLogging {
       instance.reservation
     }
 
-    // In the future the Goal should stay running even for resident tasks
-    // This is necessary right now because we need isScheduled to return false
-    // then scale check picks this up and calls LaunchQueue.Add
-    val goal = if (instance.hasReservation && updatedTask.status.condition == Condition.Reserved) Goal.Stopped else instance.state.goal
```

Comment:
Hm, should this depend on #6658?

Reply:
I think right now `isScheduled` cover this cases so we don't need to do this